pkg/kubelet/status/status_manager.go

```diff
@@ -455,6 +456,7 @@ func (m *manager) syncPod(uid types.UID, status versionedPodStatus) {
 			deleteOptions.Preconditions = metav1.NewUIDPreconditions(string(pod.UID))
 			if err = m.kubeClient.Core().Pods(pod.Namespace).Delete(pod.Name, deleteOptions); err == nil {
 				glog.V(3).Infof("Pod %q fully terminated and removed from etcd", format.Pod(pod))
+				metrics.PodDeletionLatency.Observe(metrics.SinceInMicroseconds(pod.DeletionTimestamp.Time))
```

Comment:
Caveat: DeletionTimestamp is not set by kubelet, so this will be affected by the clock difference of the machines. It would be more meaningful for kubelet itself if we measure the latency between kubelet seeing the timestamp and the actual deletion.